packages/mobx/src/types/observableobject.ts

```diff
@@ -222,7 +224,11 @@ export class ObservableObjectAdministration
             const notifySpy = __DEV__ && isSpyEnabled()
             const oldObservable = this.values_.get(key)
             const oldValue = oldObservable && oldObservable.get()
-            oldObservable && oldObservable.set(undefined)
+            // clear the observable value
+            // computed without setter throws
+            if (oldObservable && !isComputed(oldObservable)) {
```

Comment:
If without setter throws anyway, the second condition isn't needed? (assuming oldObservable.set takes care of the throwing)

Reply:
Not sure if I follow... `oldObservable` can be `computed` without defined setter - in such case `oldObservable.set(undefined)` throws, which I want to avoid as it prevents from deleting computed property.
There is no api for clearing computed's internal value, therefore it's simply ignored.
I think that setting the value to undefined is completely useless anyway (unless it's supposed to notify some listeners or something like that).

Reply:
check, misread what it was supposed to do :)